src/shell.js

```diff
@@ -175,7 +175,7 @@ var quit_ = (status, toThrow) => {
 #if SHARED_MEMORY && !MODULARIZE
 // In MODULARIZE mode _scriptName needs to be captured already at the very top of the page immediately when the page is parsed, so it is generated there
 // before the page load. In non-MODULARIZE modes generate it here.
-var _scriptName = (typeof document != 'undefined') ? document.currentScript?.src : undefined;
+var _scriptName = (typeof document != 'undefined' && document.currentScript?.tagName.toUpperCase() === 'SCRIPT') ? document.currentScript.src : undefined;
```

Comment:
Man.. this is kind of ugly, and seems like it will increase code size.  

I'm not necessarily against landing this, but I wonder if we can further limit the cases where `_scriptName` is used.. maybe we can even eliminate it somehow.

Reply:
It is pretty ugly :/ I wonder if the `.toUpperCase()` part at least is not needed? When testing, I am getting all uppercase letters in `.tagName` already. Is there a known browser where it's not in upper case, or why is the `.toUpperCase()` there?